pipeline-runner/R/gem2s-3-run_emptydrops.R

```diff
@@ -32,6 +32,30 @@ run_emptydrops <- function(input, pipeline_config, prev_out) {
   return(res)
 }
 
+determine_dynamic_threshold <- function(sample_counts, adjust = 2, min_distance_from_peak = 0.5) {
+  umi_counts <- Matrix::colSums(sample_counts)
+  log_umi_counts <- log10(umi_counts + 1)
+
+  # Calculate the kernel density estimation on the log10-transformed data with smoothing
+  density_est <- density(log_umi_counts, adjust = adjust)
+  valleys <- find_valleys(density_est, min_distance_from_peak)
+  thresholds <- 10^valleys - 1
+  return(min(thresholds))
+}
+
+find_valleys <- function(density_est, min_distance_from_peak) {
+  gradient <- diff(density_est$y)
+  sign_change <- diff(sign(gradient))
+  potential_valleys <- which(sign_change == 2) + 1
+
+  peak_location <- which.max(density_est$y)
+  peak_x_value <- density_est$x[peak_location]
+
+  # Identify valid valleys based on distance from peak
+  valid_valleys <- potential_valleys[abs(density_est$x[potential_valleys] - peak_x_value) > min_distance_from_peak]
+
+  return(density_est$x[valid_valleys])
+}
```

Comment:
Where is this from? I'm hesitant to change defaults just because they aren't working great for a single type of (less frequently used) data. If this is to go ahead, I need some evidence that this is generally fine to do for everything

Reply:
This was just a WIP, I'm going to close this PR. See [this comment here](https://github.com/hms-dbmi-cellenics/pipeline/pull/367#issuecomment-2089870208).